// tdesign-react/cjs/form/utils/index.js
/**
 * tdesign v1.15.1
 * (c) 2025 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../../_chunks/dep-8e4d656d.js');
var defineProperty = require('../../_chunks/dep-0006fcfa.js');
var toConsumableArray = require('../../_chunks/dep-e4e1901e.js');
var _hasPath = require('../../_chunks/dep-22b1740a.js');
var get = require('../../_chunks/dep-0c8c9057.js');
var isObject = require('../../_chunks/dep-fc596d16.js');
var isArray = require('../../_chunks/dep-64577888.js');
var isEmpty = require('../../_chunks/dep-6283cbd9.js');
require('../../_chunks/dep-00b49251.js');
require('../../_chunks/dep-667ac7af.js');
require('../../_chunks/dep-3398c1f2.js');
require('../../_chunks/dep-79629634.js');
require('../../_chunks/dep-25585736.js');
require('../../_chunks/dep-62e73936.js');
require('../../_chunks/dep-21ece627.js');
require('../../_chunks/dep-25e4aa84.js');
require('../../_chunks/dep-ec8d2dca.js');
require('../../_chunks/dep-e1fbe1c3.js');
require('../../_chunks/dep-255ceed8.js');
require('../../_chunks/dep-eea2872a.js');
require('../../_chunks/dep-f33c1939.js');
require('../../_chunks/dep-9e5a468d.js');
require('../../_chunks/dep-4b18243f.js');
require('../../_chunks/dep-71455db7.js');
require('../../_chunks/dep-f26edb7b.js');
require('../../_chunks/dep-bff2c990.js');

/** Used for built-in method references. */
var objectProto = Object.prototype;

/** Used to check objects for own properties. */
var hasOwnProperty = objectProto.hasOwnProperty;

/**
 * The base implementation of `_.has` without support for deep paths.
 *
 * @private
 * @param {Object} [object] The object to query.
 * @param {Array|string} key The key to check.
 * @returns {boolean} Returns `true` if `key` exists, else `false`.
 */
function baseHas(object, key) {
  return object != null && hasOwnProperty.call(object, key);
}

/**
 * Checks if `path` is a direct property of `object`.
 *
 * @static
 * @since 0.1.0
 * @memberOf _
 * @category Object
 * @param {Object} object The object to query.
 * @param {Array|string} path The path to check.
 * @returns {boolean} Returns `true` if `path` exists, else `false`.
 * @example
 *
 * var object = { 'a': { 'b': 2 } };
 * var other = _.create({ 'a': _.create({ 'b': 2 }) });
 *
 * _.has(object, 'a');
 * // => true
 *
 * _.has(object, 'a.b');
 * // => true
 *
 * _.has(object, ['a', 'b']);
 * // => true
 *
 * _.has(other, 'a');
 * // => false
 */
function has(object, path) {
  return object != null && _hasPath.hasPath(object, path, baseHas);
}

function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function getMapValue(name, formMapRef) {
  if (!formMapRef.current) return;
  var mapKeys = toConsumableArray._toConsumableArray(formMapRef.current.keys());
  var key = mapKeys.find(function (key2) {
    return String(key2) === String(name);
  });
  return formMapRef.current.get(key);
}
function objectToArray(obj) {
  var result = [];
  function traverse(current) {
    var path = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : [];
    if (isObject.isObject(current) && !isArray.isArray(current) && !isEmpty.isEmpty(current)) {
      Object.keys(current).forEach(function (key) {
        traverse(current[key], [].concat(toConsumableArray._toConsumableArray(path), [key]));
      });
    } else {
      result.push(path);
    }
  }
  traverse(obj);
  return result;
}
function calcFieldValue(name, value) {
  var isFormList = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : true;
  if (Array.isArray(name)) {
    if (isFormList) {
      var fieldValue = name.reduceRight(function (prev, curr) {
        var arr = [];
        if (/^\d+$/.test(String(curr))) arr[curr] = prev;
        return arr.length ? arr : defineProperty._defineProperty({}, curr, prev);
      }, value);
      return _objectSpread({}, fieldValue);
    }
    return name.reduceRight(function (prev, curr, currentIndex) {
      if (currentIndex === name.length - 1) {
        return defineProperty._defineProperty({}, curr, value);
      }
      return defineProperty._defineProperty({}, curr, prev);
    }, {});
  }
  return defineProperty._defineProperty({}, name, value);
}
function travelMapFromObject(obj, formMapRef, callback) {
  var _iterator = _createForOfIteratorHelper(formMapRef.current.entries()),
    _step;
  try {
    for (_iterator.s(); !(_step = _iterator.n()).done;) {
      var _step$value = slicedToArray._slicedToArray(_step.value, 2),
        mapName = _step$value[0],
        formItemRef = _step$value[1];
      if (has(obj, mapName)) {
        callback(formItemRef, get.get(obj, mapName));
      }
    }
  } catch (err) {
    _iterator.e(err);
  } finally {
    _iterator.f();
  }
}

exports.calcFieldValue = calcFieldValue;
exports.getMapValue = getMapValue;
exports.objectToArray = objectToArray;
exports.travelMapFromObject = travelMapFromObject;
