// tdesign-react/es/date-picker/panel/PanelContent.js
/**
 * tdesign v1.15.1
 * (c) 2025 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../../_chunks/dep-cb0a3966.js';
import React, { useCallback } from 'react';
import { g as getDefaultFormat } from '../../_chunks/dep-9036728c.js';
import DateHeader from '../base/Header.js';
import DatePickerTable from '../base/Table.js';
import { TimePickerPanel } from '../../time-picker/index.js';
import useConfig from '../../hooks/useConfig.js';
import useEventCallback from '../../hooks/useEventCallback.js';
import '../../_chunks/dep-eca3a3de.js';
import 'lodash-es';
import 'dayjs';
import '../../_chunks/dep-e29214cb.js';
import '../../_chunks/dep-dbf3be07.js';
import '../../_chunks/dep-f53c91cd.js';
import '../../_chunks/dep-b908e1fe.js';
import '../../_chunks/dep-87d110df.js';
import '../../_chunks/dep-026a4c6b.js';
import '../../_chunks/dep-48805ab8.js';
import '../../locale/LocalReceiver.js';
import '../../config-provider/index.js';
import '../../config-provider/ConfigProvider.js';
import '../../config-provider/ConfigContext.js';
import '../../locale/zh_CN.js';
import '../../_chunks/dep-3c9ab31a.js';
import '../../config-provider/type.js';
import '../../select/index.js';
import '../../select/base/Select.js';
import '../../_chunks/dep-6b660ef0.js';
import 'classnames';
import '../../_util/composeRefs.js';
import '../../_util/forwardRefWithStatics.js';
import 'hoist-non-react-statics';
import '../../_util/helper.js';
import '../../_util/noop.js';
import '../../_util/parseTNode.js';
import '../../common/FakeArrow.js';
import '../../hooks/useControlled.js';
import '../../hooks/useDefaultProps.js';
import '../../loading/index.js';
import '../../loading/Loading.js';
import '../../_chunks/dep-3a09424a.js';
import '../../common/Portal.js';
import 'react-dom';
import '../../hooks/useLayoutEffect.js';
import '../../loading/gradient.js';
import '../../_chunks/dep-c48e2ca1.js';
import '../../_chunks/dep-1630b9b4.js';
import '../../hooks/useDomRefCallback.js';
import '../../loading/defaultProps.js';
import '../../loading/plugin.js';
import '../../_util/react-render.js';
import '../../_chunks/dep-a74cc5e4.js';
import '../../common/PluginContainer.js';
import '../../loading/style/css.js';
import '../../loading/type.js';
import '../../select-input/index.js';
import '../../select-input/SelectInput.js';
import '../../popup/index.js';
import '../../popup/Popup.js';
import 'react-transition-group';
import '../../_util/ref.js';
import 'react-is';
import '../../_util/isFragment.js';
import '../../hooks/useAnimation.js';
import '../../hooks/useAttach.js';
import '../../hooks/useMutationObserver.js';
import '../../hooks/useLatest.js';
import '../../hooks/usePopper.js';
import '@popperjs/core';
import 'react-fast-compare';
import '../../hooks/useWindowSize.js';
import '../../popup/defaultProps.js';
import '../../popup/hooks/useTrigger.js';
import '../../_util/listener.js';
import '../../popup/utils/transition.js';
import '../../popup/PopupPlugin.js';
import '../../popup/style/css.js';
import '../../popup/type.js';
import '../../select-input/useSingle.js';
import '../../input/index.js';
import '../../input/Input.js';
import 'tdesign-icons-react';
import '../../hooks/useGlobalIcon.js';
import '../../input/InputGroup.js';
import '../../input/defaultProps.js';
import '../../input/useLengthLimit.js';
import '../../input/style/css.js';
import '../../input/type.js';
import '../../select-input/useMultiple.js';
import '../../tag-input/index.js';
import '../../tag-input/TagInput.js';
import '../../hooks/useDragSorter.js';
import '../../tag-input/defaultProps.js';
import '../../tag-input/useHover.js';
import '../../tag-input/useTagList.js';
import '../../tag/index.js';
import '../../tag/Tag.js';
import '../../_chunks/dep-3c3d29db.js';
import '../../tag/defaultProps.js';
import '../../tag/CheckTag.js';
import '../../_chunks/dep-55732ade.js';
import '../../tag/CheckTagGroup.js';
import '../../tag/style/css.js';
import '../../tag/type.js';
import '../../tag-input/useTagScroll.js';
import '../../tag-input/style/css.js';
import '../../tag-input/type.js';
import '../../select-input/useOverlayInnerStyle.js';
import '../../select-input/defaultProps.js';
import '../../select-input/style/css.js';
import '../../select-input/type.js';
import '../../select/defaultProps.js';
import '../../_chunks/dep-c7ed701c.js';
import '../../select/base/OptionGroup.js';
import '../../select/base/Option.js';
import '../../hooks/useRipple.js';
import '../../select/base/PopupContent.js';
import '../../select/hooks/usePanelVirtualScroll.js';
import '../../hooks/useVirtualScroll.js';
import '../../select/style/css.js';
import '../../select/type.js';
import '../../pagination/index.js';
import '../../pagination/Pagination.js';
import '../../input-number/index.js';
import '../../input-number/InputNumber.js';
import '../../button/index.js';
import '../../button/Button.js';
import '../../button/defaultProps.js';
import '../../button/style/css.js';
import '../../button/type.js';
import '../../input-number/useInputNumber.js';
import '../../_chunks/dep-46c9e4f5.js';
import '../../hooks/useCommonClassName.js';
import '../../input-number/defaultProps.js';
import '../../input-number/style/css.js';
import '../../input-number/type.js';
import '../../input-adornment/index.js';
import '../../input-adornment/InputAdornment.js';
import '../../input-adornment/style/css.js';
import '../../input-adornment/type.js';
import '../../pagination/hooks/useBoundaryJumper.js';
import '../../pagination/hooks/usePrevNextJumper.js';
import '../../pagination/hooks/usePageNumber.js';
import '../../pagination/hooks/useTotal.js';
import '../../pagination/validators.js';
import '../../pagination/defaultProps.js';
import '../../pagination/PaginationMini.js';
import '../../pagination/style/css.js';
import '../../pagination/type.js';
import '../base/Cell.js';
import '../../_chunks/dep-9649fdab.js';
import '../../_chunks/dep-3538034b.js';
import '../../time-picker/TimePicker.js';
import '../../_chunks/dep-f7770f5d.js';
import '../../time-picker/hooks/useTimePickerTextConfig.js';
import '../../hooks/useDebounce.js';
import '../../hooks/usePersistFn.js';
import '../../hooks/usePropsRef.js';
import '../../time-picker/TimeRangePicker.js';
import '../../range-input/index.js';
import '../../range-input/RangeInput.js';
import '../../range-input/defaultProps.js';
import '../../range-input/RangeInputPopup.js';
import '../../range-input/style/css.js';
import '../../range-input/type.js';
import '../../time-picker/panel/TimePickerPanel.js';
import '../../time-picker/defaultProps.js';
import '../../time-picker/style/css.js';
import '../../time-picker/type.js';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function PanelContent(props) {
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  var panelName = "".concat(classPrefix, "-date-picker__panel");
  var value = props.value,
    year = props.year,
    month = props.month,
    mode = props.mode,
    format = props.format,
    enableTimePicker = props.enableTimePicker,
    timePickerProps = props.timePickerProps,
    firstDayOfWeek = props.firstDayOfWeek,
    _props$partial = props.partial,
    partial = _props$partial === void 0 ? "start" : _props$partial,
    time = props.time,
    tableData = props.tableData,
    internalYear = props.internalYear,
    onJumperClick = props.onJumperClick,
    _onCellClick = props.onCellClick,
    _onCellMouseEnter = props.onCellMouseEnter,
    onCellMouseLeave = props.onCellMouseLeave,
    onTimePickerChange = props.onTimePickerChange;
  var onMonthChange = useEventCallback(props.onMonthChange);
  var onYearChange = useEventCallback(props.onYearChange);
  var _getDefaultFormat = getDefaultFormat({
      mode: mode,
      format: format,
      enableTimePicker: enableTimePicker
    }),
    timeFormat = _getDefaultFormat.timeFormat;
  var showTimePicker = enableTimePicker && mode === "date";
  var defaultTime = "00:00:00";
  var onMonthChangeInner = useCallback(function (val) {
    onMonthChange === null || onMonthChange === void 0 || onMonthChange(val, {
      partial: partial
    });
  }, [partial, onMonthChange]);
  var onYearChangeInner = useCallback(function (val) {
    onYearChange === null || onYearChange === void 0 || onYearChange(val, {
      partial: partial
    });
  }, [partial, onYearChange]);
  var onJumperClickInner = useCallback(function (_ref) {
    var trigger = _ref.trigger;
    onJumperClick === null || onJumperClick === void 0 || onJumperClick({
      trigger: trigger,
      partial: partial
    });
  }, [partial, onJumperClick]);
  return /* @__PURE__ */React.createElement("div", {
    className: "".concat(panelName, "-content")
  }, /* @__PURE__ */React.createElement("div", {
    className: "".concat(panelName, "-").concat(mode)
  }, /* @__PURE__ */React.createElement(DateHeader, {
    mode: mode,
    year: year,
    month: month,
    internalYear: internalYear,
    partial: partial,
    onMonthChange: onMonthChangeInner,
    onYearChange: onYearChangeInner,
    onJumperClick: onJumperClickInner
  }), /* @__PURE__ */React.createElement(DatePickerTable, {
    mode: mode,
    data: tableData,
    value: value,
    time: time,
    format: format,
    firstDayOfWeek: firstDayOfWeek,
    multiple: props.multiple,
    onCellClick: function onCellClick(date, _ref2) {
      var e = _ref2.e;
      return _onCellClick === null || _onCellClick === void 0 ? void 0 : _onCellClick(date, {
        e: e,
        partial: partial
      });
    },
    onCellMouseEnter: function onCellMouseEnter(date) {
      return _onCellMouseEnter === null || _onCellMouseEnter === void 0 ? void 0 : _onCellMouseEnter(date, {
        partial: partial
      });
    },
    onCellMouseLeave: onCellMouseLeave
  })), showTimePicker && /* @__PURE__ */React.createElement("div", {
    className: "".concat(panelName, "-time")
  }, /* @__PURE__ */React.createElement("div", {
    className: "".concat(panelName, "-time-viewer")
  }, time || defaultTime), /* @__PURE__ */React.createElement(TimePickerPanel, _objectSpread({
    key: partial,
    position: partial,
    format: timeFormat,
    value: time || defaultTime,
    onChange: onTimePickerChange,
    isShowPanel: props.popupVisible
  }, timePickerProps))));
}

export { PanelContent as default };
